sgt-back-end/index.js

```diff
@@ -0,0 +1,177 @@
+    })
+    .catch(err => {
+      // the query failed for some reason
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades', (req, res, next) => {
+  const sql2 = `
+    select *
+    from "grades"
+  `;
+  db.query(sql2)
+    .then(result => {
+      // res.json(result.rows);
+      const fullTable = result.rows;
+      res.status(200).json(fullTable);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An error happened but idk exactly what...'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  var NAM = req.body.name;
+  var COU = req.body.course;
+  var GRA = req.body.grade;
+  var numGrade = parseInt(GRA);
+  if (isNaN(numGrade)) {
+    res.status(400).json({
+      error: 'Lets, uhhh, maybe supply an actual integer grade value, guys ....'
+    });
+  } else {
+    var sql3 = `
+  insert into "grades" ("name", "course", "grade")
+  values('${NAM}', '${COU}', ${numGrade})
+  returning *;
+  `;
+    db.query(sql3)
+      .then(result => {
+        const student = result.rows;
+        res.status(201);
+        res.json(student);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      });
+  }
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+  }
+  var NAM = req.body.name;
+  var COU = req.body.course;
+  var GRA = req.body.grade;
+  var numGrade = parseInt(GRA);
+  if (isNaN(numGrade)) {
+    res.status(400).json({
+      error: 'Lets, uhhh, maybe supply an actual integer grade value, guys ....'
+    });
+  } else {
+    var sql4 = `
+    update "grades"
+    set "name" = '${NAM}',
+        "course" = '${COU}',
+        "grade" = ${numGrade}
+    where "gradeId" = ${gradeId}
+    returning *;
+    `;
+    db.query(sql4)
+      .then(result => {
+        const student = result.rows;
+        if (!student) {
+          res.status(404).json({
+            error: `Cannot find grade with "gradeId" ${gradeId}`
+          });
+        } else {
+          res.status(200);
+          res.json(student);
+        }
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occurred.'
+        });
+      });
+  }
+});
+
+app.delete('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = parseInt(req.params.gradeId, 10);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    return res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+  }
+  var sql5 = `
+    delete from "grades"
+    where "gradeId" = ${gradeId}
+    returning *;
+    `;
```

Comment:
# ☠️ SQL Injection Vulnerability
You need to put your values into an array, and put placeholders into the query instead `$1`. The array of values is the second argument to `db.query()` check the docs of `node-postgres` regarding parameterized queries.